Reorganize incubation repo to allow hosting multiple projects

alexnj opened the pull request:
* A repo-wide readme with an overview and links to all associated documentation/artifacts so far.
 * A more flexible directory structure that would allow us to host all prototypes/demos and upcoming SDKs.
   * `implementations/` containing all code implementations. 
     * `cf-worker-prototype` the original ab worker prototype that runs on CloudFlare Edge.
     * _(post-merge)_ `atticandbutton-ab-demo` the 2023 March demo that rewrites a Shopify site.
     * _(post-merge)_ `npm` the NPM library/sdk we're planning to extract the current implementation into.
   * `specs/` containing explainers and other docs.

micahheineckoptimizely commented:
I would like to strongly differentiate between tools (libraries, SDKs) that we expect people to be able to import and use directly, compare to demos (examples usages of the tools). Something like:

- `specs/` containing explainers and other docs.
- `tools/` containing SDKs, libraries, etc
  -  _(post-merge)_ npm the NPM library/sdk we're planning to extract the current implementation into.
  -  _(post-merge)_ the applicator code from the current `cf-worker-prototype` 
- `demos`
  - `cf-worker-prototype` the original ab worker prototype, _(post-merge)_ importing the applicator code from `tools/edge-applicator`, that runs on CloudFlare Edge.
  - _(post-merge)_ `atticandbutton-ab-demo` the 2023 March demo that rewrites a Shopify site.

alexnj commented:
@micahheineckoptimizely I updated the structure to reflect your comment. I renamed `tools` to `sdks`, and for differentiating demos/prototypes, we could just include `-prototype` or `-demo` in the folder name than increasing path depth.

micahheineckoptimizely reviewed (approved):
LGTM, thanks!
